assets/src/edit-story/components/library/panes/text/textPane.js

```diff
@@ -32,17 +33,83 @@ import { Section, MainButton, SearchInput } from '../../common';
 import { FontPreview } from '../../text';
 import useLibrary from '../../useLibrary';
 import { Pane } from '../shared';
+import { dataFontEm } from '../../../../units';
+import { useHistory } from '../../../../app/history';
+import getInsertedElementSize from '../../../../utils/getInsertedElementSize';
+import { PAGE_HEIGHT } from '../../../../constants';
 import paneId from './paneId';
 import { PRESETS, DEFAULT_PRESET } from './textPresets';
 
 const SectionContent = styled.p``;
 
+const POSITION_MARGIN = dataFontEm(1);
+const TYPE = 'text';
+
 function TextPane(props) {
   const { insertElement } = useLibrary((state) => ({
     insertElement: state.actions.insertElement,
   }));
+  const {
+    state: { versionNumber },
+  } = useHistory();
+
   const { showTextSets, showTextAndShapesSearchInput } = useFeatures();
 
+  const lastPreset = useRef(null);
+
+  const getPosition = useCallback(
+    (element) => {
+      const { y } = element;
+      // If the difference between the new and the previous version number is not 1,
+      // the preset wasn't clicked right after the previous.
+      if (
+        !lastPreset.current ||
+        versionNumber - lastPreset.current.versionNumber !== 1
+      ) {
```

Comment:
It's a bit tricky to rely on version number only accumulating once. For instance, auto-save will probably update history, which might conflict with this. The "good" solution would be to only set `lastPreset` after the history has updated and then clear it if history updates again in-between. But with the current structure that's not really possible. We don't have a way to say "and after the next update, do this".

Reply:
Hmm, auto-save nor save shouldn't really update the version number/history though -- it should only update the saved version number. If auto-save (or save) do that then there's an issue with history, perhaps related to #4094 

`versionNumber - lastPreset.current.versionNumber !== 1` is actually the check to see if the history has updated meanwhile -- in any other case then the difference being `1` something other than adding a preset has happened.

Just tested that currently saving the story seems to modify the history indeed (e.g. saving and then clicking Undo does nothing, indicating that the save itself has modified something) -- this would need investigation, will add a note to #4094.

Reply:
Changed the logic a bit in 1704efb -- let me know if it's a bit more clear now. However, note that due to the history bug it still stops the consecutive ordering when saving/auto-saving. 🤷 
I think we can live with that until the history bug gets fixed though. Debugged it a bit and saving, or loading a new story indeed seems to create a new (identical to the previous) history entry into history -- perhaps there's object mutation happening somewhere instead of returning the original object.

Thoughts?

Reply:
I think the new structure works well.

The "bug" is due to us [post-save updating the story with the correct link and status of the story](https://github.com/google/web-stories-wp/blob/37bc1c758d99f4d13015374e0f67fafcd160d6ff/assets/src/edit-story/app/story/actions/useSaveStory.js#L78-L82) - e.g. if the user has changed permalink settings since last saving, the permalink will come back changed when saving is complete.

The bug could then be considered that history is reported as changed, even though no attribute actually updated. But saving can and should be able to update the story and thus progress history, due to the above possible changes happening outside the editor.